src/code/z_demo.c

```diff
@@ -1232,684 +1406,776 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
-                    if ((temp == 0.0f) && (gSaveContext.entranceIndex == ENTR_CHAMBER_OF_THE_SAGES_0)) {
+
+                if (cmd->type == CS_TRANS_GRAY_FILL) {
+                    play->envCtx.screenFillColor[3] = 255.0f * lerp;
+
+                    if ((lerp == 0.0f) && (gSaveContext.entranceIndex == ENTR_CHAMBER_OF_THE_SAGES_0)) {
                         Audio_PlaySfxGeneral(NA_SE_SY_WHITE_OUT_S, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
                                              &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
-                    } else if ((temp == 0.0f) &&
+                    } else if ((lerp == 0.0f) &&
                                ((gSaveContext.entranceIndex == ENTR_TEMPLE_OF_TIME_0) ||
                                 (gSaveContext.entranceIndex == ENTR_HYRULE_CASTLE_0) ||
                                 (gSaveContext.entranceIndex == ENTR_GREAT_FAIRYS_FOUNTAIN_SPELLS_0))) {
                         Audio_PlaySfxGeneral(NA_SE_EV_WHITE_OUT, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
                                              &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
-                    } else if ((temp == 0.0f) && (play->sceneId == SCENE_INSIDE_GANONS_CASTLE)) {
+                    } else if ((lerp == 0.0f) && (play->sceneId == SCENE_INSIDE_GANONS_CASTLE)) {
                         func_800788CC(NA_SE_EV_WHITE_OUT);
                     }
                 } else {
-                    play->envCtx.screenFillColor[3] = (1.0f - temp) * 255.0f;
+                    play->envCtx.screenFillColor[3] = (1.0f - lerp) * 255.0f;
                 }
                 break;
 
-            case 2:
-            case 6:
+            case CS_TRANS_BLUE_FILL:
+            case CS_TRANS_BLUE_UNFILL:
                 play->envCtx.screenFillColor[0] = 0;
                 play->envCtx.screenFillColor[1] = 0;
                 play->envCtx.screenFillColor[2] = 255;
-                if (cmd->base == 2) {
-                    play->envCtx.screenFillColor[3] = 255.0f * temp;
+
+                if (cmd->type == CS_TRANS_BLUE_FILL) {
+                    play->envCtx.screenFillColor[3] = 255.0f * lerp;
                 } else {
-                    play->envCtx.screenFillColor[3] = (1.0f - temp) * 255.0f;
+                    play->envCtx.screenFillColor[3] = (1.0f - lerp) * 255.0f;
                 }
                 break;
 
-            case 3:
-            case 7:
+            case CS_TRANS_RED_FILL:
+            case CS_TRANS_RED_UNFILL:
                 play->envCtx.screenFillColor[0] = 255;
                 play->envCtx.screenFillColor[1] = 0;
                 play->envCtx.screenFillColor[2] = 0;
-                if (cmd->base == 3) {
-                    play->envCtx.screenFillColor[3] = (1.0f - temp) * 255.0f;
+
+                if (cmd->type == CS_TRANS_RED_FILL) {
+                    play->envCtx.screenFillColor[3] = (1.0f - lerp) * 255.0f;
```

Comment:
I think `CS_TRANS_RED_FILL` and `CS_TRANS_RED_UNFILL` are swapped

same with green and black